src/modules/fundingManager/bondingCurve/abstracts/BondingCurveBase_v1.sol

```diff
@@ -125,28 +117,66 @@ abstract contract BondingCurveBase_v1 is IBondingCurveBase_v1, Module_v1 {
 
     /// @inheritdoc IBondingCurveBase_v1
     function calculatePurchaseReturn(uint _depositAmount)
-        external
+        public
         virtual
         returns (uint mintAmount)
     {
-        return _calculatePurchaseReturn(_depositAmount);
+        _validateDepositAmount(_depositAmount);
+        // Get protocol fee percentages
+        (
+            /* collateralreasury */
+            ,
+            /* issuanceTreasury */
+            ,
+            uint collateralBuyFeePercentage,
+            uint issuanceBuyFeePercentage
+        ) = _getFunctionFeesAndTreasuryAddresses(
+            bytes4(keccak256(bytes("_buyOrder(address, uint, uint)")))
+        );
+
+        // Deduct protocol and project buy fee from collateral, if applicable
+        (_depositAmount, /* protocolFeeAmount */, /* workflowFeeAmount */ ) =
+        _calculateNetAndSplitFees(
+            _depositAmount, collateralBuyFeePercentage, buyFee
+        );
+
+        // Get issuance token return from formula and deduct protocol buy fee, if applicable
+        (mintAmount, /* protocolFeeAmount */, /* workflowFeeAmount */ ) =
+        _calculateNetAndSplitFees(
+            _issueTokensFormulaWrapper(_depositAmount),
+            issuanceBuyFeePercentage,
+            0
+        );
+
+        // Return expected purchase return amount
+        //return mintAmount;
     }
 
     /// @inheritdoc IBondingCurveBase_v1
     function withdrawProjectCollateralFee(address _receiver, uint _amount)
-        external
+        public
         virtual
         validReceiver(_receiver)
         onlyOrchestratorOwner
     {
-        _withdrawProjectCollateralFee(_receiver, _amount);
+        if (_amount > projectCollateralFeeCollected) {
+            revert Module__BondingCurveBase__InvalidWithdrawAmount();
+        }
+
+        projectCollateralFeeCollected -= _amount;
+
+        __Module_orchestrator.fundingManager().token().safeTransfer(
+            _receiver, _amount
+        );
+
+        emit ProjectCollateralFeeWithdrawn(_receiver, _amount);
     }
 
     //--------------------------------------------------------------------------
     // Public Functions
 
     /// @notice Returns the address of the issuance token
-    function getIssuanceToken() public view virtual returns (address) {
+    function getIssuanceToken() external view virtual returns (address) {
```

Comment:
Why are most function visibilities changes from external to public, but this one the other way around? Just curious

Reply:
Since the contract can access the variable directly I thought it wasn't needed to make it public. The other changes from external to public were mostly because we are now calling the functions (f.e. buyFor) from the contract itself.